bullet-featherstone/src/SDFFeatures.cc

```diff
@@ -254,23 +260,30 @@ std::optional<Structure> ValidateModel(const ::sdf::Model &_sdfModel)
           }
 
           rootLink = link;
-          continue;
         }
-
-        linkIndex[link] = linkIndex.size();
-        flatLinks.push_back(link);
       }
 
       return true;
     };
 
-  if (!flattenLinks(_sdfModel))
+  if (rootLink == nullptr && !findRootLink(_sdfModel)) {
     return std::nullopt;
+  }
 
   for (std::size_t i = 0; i < _sdfModel.ModelCount(); ++i)
   {
-    if (!flattenLinks(*_sdfModel.ModelByIndex(i)))
+    if (rootLink != nullptr) {
+      break;
+    }
+    if (!findRootLink(*_sdfModel.ModelByIndex(i))) {
       return std::nullopt;
+    }
```

Comment:
can you add comments for this for loop? It looks like it searches for a root link in nested models if one is not already found

Reply:
added comment. [e689f8b](https://github.com/gazebosim/gz-physics/pull/562/commits/e689f8b9513c7aa3e94eea73b2979f883699cf3d)